### ISSUE TITLE
visdom 无法显示，是代码问题么？同时提示这个错误！

### ISSUE THREAD

**Issue by loveproxy** [chenyuntc/simple-faster-rcnn-pytorch#85]
Without the incoming socket you cannot receive events from the server or register event handlers to your Visdom client

**Comment by Bigwode** [comment]
because you not run this command before run your train code 
nohup python -m visdom.server &

**Comment by gmt710** [comment]
> because you not run this command before run your train code
> nohup python -m visdom.server &

@Bigwode Hello,I refer to your solution, but the problem is still there.

**Comment by gmt710** [comment]
> Without the incoming socket you cannot receive events from the server or register event handlers to your Visdom client

@loveproxy ,Hi,have you solved this problem? I met this problem too in my server.

**Comment by kangkang59812** [comment]
I met this too. But When I waited for a minute , the visdom begined to show the curve and data.

**Comment by yuanzhoulvpi2017** [comment]
实际上，你只要等待一会就可以解决了，现在visdom已经没有问题了，很多人就是不知道等一会，（如果是chrome浏览器，可以在左下角看到加载项目）

**Comment by hktxt** [comment]
not working... 浏览器左下角 加载失败

WARNING:root:Setting up a new session...
WARNING:visdom:Without the incoming socket you cannot receive events from the server or register event handlers to your Visdom client.

**Comment by fmscole** [comment]
1)run this command before run your train code:
```
nohup python -m visdom.server &
```
2)Environment:select 'faster-rcnn', not 'main'

**Comment by ZhenhuiTang** [comment]
how to visualize the training process? 
when i run: nohup python -m visdom.server &,  return: nohup: ignoring input and appending output to 'nohup.out'
then I run the command: python train.py train
it shows :
....
 'weight_decay': 0.0005}
==========end============
load data
model construct completed
Setting up a new session...
Without the incoming socket you cannot receive events from the server or register event handlers to your Visdom client.
1420it [05:13,  4.83it/s]
...

I can't see train loss...

**Comment by yuanzhoulvpi2017** [comment]
为什么今天还有人用visdom，现在pytorch已经可以用tensorboard了

**Comment by bubblehhhahaha** [comment]
> how to visualize the training process?
> when i run: nohup python -m visdom.server &, return: nohup: ignoring input and appending output to 'nohup.out'
> then I run the command: python train.py train
> it shows :
> ....
> 'weight_decay': 0.0005}
> ==========end============
> load data
> model construct completed
> Setting up a new session...
> Without the incoming socket you cannot receive events from the server or register event handlers to your Visdom client.
> 1420it [05:13, 4.83it/s]
> ...
> 
> I can't see train loss...

**Comment by bubblehhhahaha** [comment]
> how to visualize the training process?
> when i run: nohup python -m visdom.server &, return: nohup: ignoring input and appending output to 'nohup.out'
> then I run the command: python train.py train
> it shows :
> ....
> 'weight_decay': 0.0005}
> ==========end============
> load data
> model construct completed
> Setting up a new session...
> Without the incoming socket you cannot receive events from the server or register event handlers to your Visdom client.
> 1420it [05:13, 4.83it/s]
> ...
> 
> I can't see train loss...

can  you  solve this problem? it still downloads something but I donnot known what is it

**Comment by Leonliou** [comment]
> > how to visualize the training process?
> > when i run: nohup python -m visdom.server &, return: nohup: ignoring input and appending output to 'nohup.out'
> > then I run the command: python train.py train
> > it shows :
> > ....
> > 'weight_decay': 0.0005}
> > ==========end============
> > load data
> > model construct completed
> > Setting up a new session...
> > Without the incoming socket you cannot receive events from the server or register event handlers to your Visdom client.
> > 1420it [05:13, 4.83it/s]
> > ...
> > I can't see train loss...
> 
> can you solve this problem? it still downloads something but I donnot known what is it

and now? i meet too

**Comment by 183731491** [comment]
> Without the incoming socket you cannot receive events from the server or register event handlers to your Visdom client

hi. i met the same problem.have you fix it?

**Comment by shida666** [comment]
> 1)run this command before run your train code:
> 
> ```
> nohup python -m visdom.server &
> ```
> 
> 2)Environment:select 'faster-rcnn', not 'main'

hey bro,i use the original code like  "nohup python -m visdom.server &

python train.py train --env='fasterrcnn-caffe' --plot-every=100 --caffe-pretrain"  but i can't open "http://<ip>:8097"  can you help me ?

**Comment by csg1111** [comment]
没有解决，请教一下，demo不显示啊

**Comment by csg1111** [comment]
> > Without the incoming socket you cannot receive events from the server or register event handlers to your Visdom client
> 
> hi. i met the same problem.have you fix it?

你解决了吗？

**Comment by kaizhenzhang** [comment]
> > Without the incoming socket you cannot receive events from the server or register event handlers to your Visdom client
> 
> @loveproxy ,Hi,have you solved this problem? I met this problem too in my server.

Hi，have you solved this problem now？ Please tell me why,thanks.

**Comment by code-wd** [comment]
> > > Without the incoming socket you cannot receive events from the server or register event handlers to your Visdom client
> > 
> > 
> > hi. i met the same problem.have you fix it?
> 
> 你解决了吗？

- 让子弹飞一会
- 我这个子弹飞了十多分钟才显示

**Comment by gmt710** [comment]
加上下面这句就行了
python -m visdom.server &
